### PR TITLE
skl: replace AtomicU32 with AtomicPtr

### PR DESCRIPTION
**Pull request by Fullstop000**
This PR uses `AtomicPtr` instead of `AtomicU32` in the skiplist, which simplifies some handing
Signed-off-by: Fullstop000 <fullstop1005@gmail.com>

### CODE REVIEW

**Comment by Fullstop000** [comment]
@skyzh @BusyJay @Connor1996  PTAL

**Review by BusyJay — COMMENTED** [review]
Another problem

File: skiplist/src/list.rs [hunk 1]
```diff
@@ -17,28 +17,29 @@ pub struct Node {
     key: Bytes,
     value: Bytes,
     height: usize,
-    tower: [AtomicU32; MAX_HEIGHT as usize],
+    tower: [AtomicPtr<Self>; MAX_HEIGHT as usize],
```
[the remark this answers is not in the document]
Comment (Fullstop000) [reply skiplist/src/list.rs:20]: It may have some impact (not good also not bad). I'll do a more concrete investigation if possible.

**Comment by zhangjinpeng1987** [comment]
@Fullstop000 can we close this stale PR?

**Comment by zhangjinpeng1987** [comment]
@Fullstop000 can we close this stale PR since the arena can grow, if use not offset but ptr, it will hard to memcpy when grow the arena.

**Pull request closed without merging**